### PR TITLE
Session Serialization fix for 2.28

### PR DESCRIPTION
**Pull request by wpride**
2.28 version of https://github.com/dimagi/commcare-android/pull/1347

@phillipm I slightly rewrote the continue logic for the sync and query requests

### CODE REVIEW

**Comment by amstone326** [comment]
@wpride Haven't looked at this yet just because it's not labeled with anything. Is it good to go?

**Comment by wpride** [comment]
@amstone326 yes

File: app/src/org/commcare/android/database/user/models/SessionStateDescriptor.java [hunk 1]
```diff
@@ -82,49 +83,44 @@ public String getSessionDescriptor() {
...
         int current = 0;
         while (current < tokenStream.length) {
             String action = tokenStream[current];
             if (action.equals(SessionFrame.STATE_COMMAND_ID)) {
                 session.setCommand(tokenStream[++current]);
-            } else if (action.equals(SessionFrame.STATE_DATUM_VAL) || action.equals(SessionFrame.STATE_DATUM_COMPUTED)) {
+            } else if (action.equals(SessionFrame.STATE_DATUM_VAL) ||
+                    action.equals(SessionFrame.STATE_DATUM_COMPUTED)) {
                 session.setDatum(tokenStream[++current], tokenStream[++current]);
-            } else if (action.equals(SessionFrame.STATE_SYNC_REQUEST)
```
Comment (amstone326) [inline app/src/org/commcare/android/database/user/models/SessionStateDescriptor.java]: @phillipm Removing this seems like a change in functionality?

Comment (phillipm) [inline app/src/org/commcare/android/database/user/models/SessionStateDescriptor.java]: Yes. This code was introduced by https://github.com/dimagi/commcare-android/pull/1059/files#diff-cd3d036d05167f3e910a1381ea329b91R121 (in 2.28) and wasn't well thought out on my part, so I think these new changes are the right thing now.

File: unit-tests/src/org/commcare/android/database/user/models/SessionStateDescriptorTests.java [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+        session.clearAllState();
+        session.setCommand("m0");
+        session.setCommand("m0-f0");
+        session.setDatum("case_id", "case_two");
+        session.setComputedDatum(session.getEvaluationContext());
+
+        serializeSessionOutToDescriptor(session);
+    }
+
+    private static void serializeSessionOutToDescriptor(CommCareSession session) {
```
Comment (amstone326) [inline unit-tests/src/org/commcare/android/database/user/models/SessionStateDescriptorTests.java]: @phillipm I'm confused as to what this is testing. What are `newSessionWrapper` and `originalSessionWrapper`?

Comment (phillipm) [inline unit-tests/src/org/commcare/android/database/user/models/SessionStateDescriptorTests.java]: It tests 
 - A: building session descriptors from session wrappers
 - B: building session wrappers from session descriptors.
It makes sure that doing A(B(A(session))) gives you back the same session descriptor that you started with.

**Pull request merged**